pybamm/discretisations/discretisation.py

```diff
@@ -209,11 +206,32 @@ def create_jacobian(self, model):
 
         Parameters
         ----------
-        model : :class:`pybamm.BaseModel`
-            Model to dicretise. Must have attributes rhs, initial_conditions and
+        model : :class:`pybamm.BaseModel` (or subclass)
+            Discretised model. Must have attributes rhs, initial_conditions and
             boundary_conditions (all dicts of {variable: equation})
         """
-        # TO DO: create jacobian by differentiating tree wrt StateVector
+        # Get number points in model
+        N = model.concatenated_initial_conditions.shape[0]
+
+        # Create StateVector to differentiate model with respect to
+        y = pybamm.StateVector(slice(0, N))
+
+        # Differentiate concatenated rhs and algebraic equations w.r.t. the
+        # entire StateVector
+        jac_rhs = dict.fromkeys(model.rhs.keys())
+        for eqn_key, eqn in model.rhs.items():
+            jac_rhs[eqn_key] = eqn.jac(y)
+
+        jac_algebraic = dict.fromkeys(model.algebraic.keys())
+        for eqn_key, eqn in model.algebraic.items():
+            jac_algebraic[eqn_key] = eqn.jac(y)
+
+        jacobian = self.sparse_stack(*jac_rhs.values(), *jac_algebraic.values())
+
+        def jacfn(t, y):
+            return jacobian.simplify().evaluate(t, y)
+
+        model.jacobian = jacfn
```

Comment:
are we using this function at all now? if not, ok to remove it entirely

Reply:
yeah this isn't used an was a hangover from before. have removed.